bundles/io/org.eclipse.smarthome.io.security/src/main/java/org/eclipse/smarthome/io/security/api/SecureSocketCapabilities.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.security.api;
+
+/**
+ * A set of default capabilities. Implement SecureSocketCapability and define your own for additional capabilities.
+ */
+public enum SecureSocketCapabilities implements SecureSocketCapability {
+    /** Capability: Request a TLS capable secure socket implementation */
+    TLS,
+    /**
+     * Capability: Request a DTLS capable secure socket implementation.
+     * On Java 8 this is usually a Scandium DtlsConnector. On Java 9 it is a SSLContext.
```

Comment:
If we have to make the distinction between `SSLCONTEXT` and `SCANDIUM_CONNECTOR`, why do you duplicate the information by also offering `DTLS` here?

Reply:
I'd like to be the first user of this API in another binding and I plan to support Java9 SSLContexts as well as a DtlsConnector. Preferably it is a SSLContext of course. So I would only require the capability "DTLS" and don't care which implementation is actually returned. Other bindings might do this different and definitely want a DtlsConnector. They would request the SCANDIUM_CONNECTOR+DTLS capability.

Reply:
Which of the possible options below is a user of the API required to ask for?
* `DTLS` AND `SSLContext` OR `DTLS` AND `SCANDIUM_CONNECTOR`
* `SSLContext` OR `SCANDIUM_CONNECTOR`
* just `DTLS` and depending on the java runtime version you dynamically get a `SSLContext` OR `SCANDIUM_CONNECTOR`

Reply:
It is unfortunately that complicated because Java 8 installations need to be considered. The API is more generic therefore. The implementation of this API that ESH provides would return a SSLContext only, to keep dependencies low. On Java 8, if you request `TLS`, you get something back, if you request `DTLS`, no provider can be returned. You can optionally require `SSLContext` to make sure, it is always a SSLContext. Because:

Another bundle can provide more options e.g. Scandium Connectors additionally. It is more or less undefined which secure socket provider will be choosen, if you just request `DTLS`.

I have adjusted the comments in code to explain the flags a little more.